src/app/catalog/containers/equipment-item/equipment-item.component.ts

```diff
@@ -39,6 +51,42 @@ export class EquipmentItemComponent implements OnInit {
     });
   }
 
+  getSelectedRentPeriod(equipmentId?: number, maxRentalPeriod?: number) {
+    this.controller
+      .getRentPeriods(equipmentId, maxRentalPeriod)
+      .pipe(untilDestroyed(this))
+      .subscribe((period) => {
+        this.selectedRentPeriod = <UnavailableDates | null>period;
+        this.cdr.markForCheck();
+      });
+  }
+
+  createOrder(selectedRentPeriod: UnavailableDates, equipmentId?: number) {
+    if (equipmentId) {
```

Comment:
if (equpment) return:
this way is called erly retrun form function. there is pople who doesn't like this but I think it makes the code more readable.

Reply:
done